airflow/serialization/serde.py

```diff
@@ -72,89 +70,104 @@ def decode(d: dict[str, str | int | T]) -> tuple:
     return d[CLASSNAME], d[VERSION], d.get(DATA, None)
 
 
-def serialize(o: object, depth: int = 0) -> U | None:
+@dataclasses.dataclass()
+class _Serializer:
```

Comment:
Do not put this in a class. Classes are overhead on the stack and significantly slow down serialization/deserialization. The code was implemented on purpose o be as high up in the chain as possible.